zepid/causal/doublyrobust/TMLE.py

```diff
@@ -77,35 +77,42 @@ def exposure_model(self, model, custom_model=None, bound=False, print_results=Tr
         """
         self._exp_model = self._exposure + ' ~ ' + model
 
-        # Base logistic regression model to generated predicted probabilities
+        # Step 3) Estimation of g-model (exposure model)
         if custom_model is None:
             fitmodel = propensity_score(self.df, self._exp_model, print_results=print_results)
-            self.gW = fitmodel.predict(self.df)
-            if bound is not False:  # Bounding predicted probabilities if requested
-                self.gW = self._bounding(self.gW, bounds=bound)
+            self.g1W = fitmodel.predict(self.df)
 
         # User-specified prediction model
         else:
-            try:  # This two-stage 'try' filters whether the data needs an intercept, then has the predict() attr
-                data = patsy.dmatrix(model, self.df)
-                try:
-                    self.gW = custom_model.predict(data)
-                    if bound is not False:  # Bounding predicted probabilities if requested
-                        self.gW = self._bounding(self.gW, bounds=bound)
-                except AttributeError:
-                    raise AttributeError("custom_model does not have the 'predict()' attribute")
-            except ValueError:
-                data = patsy.dmatrix(model+' - 1', self.df)
-                try:
-                    if hasattr(custom_model, 'predict_proba'):
-                        self.gW = custom_model.predict_proba(data)[:, 1]
+            if hasattr(custom_model, 'fit'):
+                # Supports sklearn and supylearner
+                if {'X', 'y'}.issubset(set(inspect.getfullargspec(custom_model.fit)[0])):
```

Comment:
this looks hairy - what is it doing?

Reply:
mm I see. I think a more Pythonic option (rather than this meta programming approach) is to try-catch it:
```python
data = patsy.dmatrix(model + ' - 1', self.df)
try:
    fm = custom_model.fit(X=data, y=self.df[self._outcome])
except TypeError:
   raise TypeError("Currently custom_...
if hasattr(fm, 'predict_proba'):
  ...
```